tests/tests/poa.rs

```diff
@@ -85,6 +90,73 @@ async fn can_get_sealed_block_from_poa_produced_block() {
         .expect("failed to verify signature");
 }
 
+#[tokio::test]
+async fn can_get_sealed_block_from_poa_produced_block_when_signing_with_kms() {
```

Comment:
I think it would make sense to have a test for if you don't specify either a key or kms. Show that it fails to produce blocks somehow.

Reply:
Seems too hard to test, not doing this for now.